src/screens/ExchangeModal.js

```diff
@@ -275,33 +278,33 @@ export default function ExchangeModal({
 
   // Set default gas limit
   useEffect(() => {
-    if (isEmpty(prevGasPrices) && !isEmpty(gasPrices)) {
+    if (isEmpty(prevGasPrices) && !isEmpty(gasFeeParamsBySpeed)) {
```

Comment:
just a note: this is comparing two different objects (`gasFeesBySpeed` vs `gasFeeParamsBySpeed`)